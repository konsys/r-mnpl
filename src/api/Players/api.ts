import { client } from "../../http/client";
import { IPlayer } from "../../types/types";

const profileUrl = `/users/profile`;

export async function fetchPlayers(params?: any): Promise<IPlayer[]> {
  return await (await client.get(profileUrl, params)).data;
}
