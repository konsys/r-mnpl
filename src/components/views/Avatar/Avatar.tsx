import React from "react";
import { Params } from "../../../config/params";

export interface AvatarProps {
  money: number;
  remainTime: number;
  isVip: boolean;
  avatar: string;
  name: string;
}
export const Avatar = (props: AvatarProps) => (
  <div className="table-body-players-card-body">
    <div className="table-body-players-card-body-avatar">
      <div>
        <div
          style={{ backgroundImage: `url(${Params.BASE_URL}${props.avatar})` }}
        />
      </div>
    </div>
    <div className="table-body-players-card-body-nick">
      {props.isVip && <div className="_vip" />}
      <div className="_nick">
        <div>{props.name}</div>
      </div>
      <div className="_muted" style={{ display: "none" }} />
      <div className="_ignore" style={{ display: "none" }} />
    </div>
    <div className="table-body-players-card-body-money">{props.money}</div>
    <div
      className="table-body-players-card-body-timer"
      style={{ display: "block" }}
    >
      <svg
        xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 100 100"
        preserveAspectRatio="none"
      >
        <circle cx="50" cy="50" r="46" transform="rotate(-90 50 50)" />
      </svg>
      <div>{props.remainTime}</div>
    </div>
  </div>
);
