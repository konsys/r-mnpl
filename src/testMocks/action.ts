import { ICurrentAction } from "stores/Board/ActionStore";
import {
  BoardAction,
  IncomeMessageType,
  OutcomeMessageType,
} from "types/types";

export const testDoNothingAction: ICurrentAction = {
  actionId: "testActionId",
  event: {
    action: {
      _id: "testID",
      isModal: false,
      userId: 1,
      type: IncomeMessageType.DO_NOTHING,
    },
  },
};

export const testRollDicesModal: BoardAction = {
  _id: "testBoardAction",
  type: IncomeMessageType.INCOME_ROLL_DICES_MODAL,

  userId: 1,
  title: "testTitleRollDices",
  text: "testTextRollDices",
  isModal: true,
};
export const testRollDicesClicked: BoardAction = {
  _id: "testBoardAction",
  type: OutcomeMessageType.OUTCOME_ROLL_DICES_CLICKED,
  userId: 1,
  isModal: false,
};
