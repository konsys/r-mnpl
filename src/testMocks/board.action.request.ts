import { IBoardActionRequest, OutcomeMessageType } from "types/types";

export const testBoardActionRequest:IBoardActionRequest = {
    gameId:'testGameId',
    action:OutcomeMessageType.OUTCOME_ROLL_DICES_CLICKED
}