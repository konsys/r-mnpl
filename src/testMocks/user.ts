import { IPlayer, IUser } from "types/types";
import { PlayerRoomStatus, RoomPlayer } from "stores/Game/Rooms/RoomsModel";

export const testPlayer1: IPlayer = {
  frags: "",
  userId: 1,
  money: 15000,
  vip: true,
  registrationType: "none",
  name: "IVAN",
  team: undefined,
  avatar: "testAvatar1",
  createdAt: new Date(),
  updatedAt: new Date(),
  isActive: false,
  isBlocked: true,
  isActing: true,
  gameId: "",
  doublesRolledAsCombo: 0,
  jailed: 0,
  unjailAttempts: 0,
  meanPosition: 0,
  creditPayRound: false,
  creditNextTakeRound: 0,
  score: 0,
  additionalTime: 0,
  timeReduceLevel: 0,
  creditToPay: 0,
  canUseCredit: false,
  moveOrder: 0,
};

export const testPlayer2: IPlayer = {
  frags: "",
  userId: 1,
  money: 15000,
  vip: true,
  registrationType: "none",
  name: "DIMAN",
  team: undefined,
  avatar: "testAvatar2",
  createdAt: new Date(),
  updatedAt: new Date(),
  isActive: false,
  isBlocked: true,
  isActing: false,
  gameId: "",
  doublesRolledAsCombo: 0,
  jailed: 0,
  unjailAttempts: 0,
  meanPosition: 0,
  creditPayRound: false,
  creditNextTakeRound: 0,
  score: 0,
  additionalTime: 0,
  timeReduceLevel: 0,
  creditToPay: 0,
  canUseCredit: false,
  moveOrder: 0,
};

export const testUser: IUser = {
  userId: 1,
  vip: false,
  name: "IVAN",
  isActive: true,
  isBlocked: false,
};
export const test5User: IUser = {
  userId: 5,
  vip: false,
  name: "NIKOLA",
  isActive: true,
  isBlocked: false,
};
export const testAvatarUser: IUser = {
  avatar: "/test/avatar.jpg",
  userId: 2,
  vip: false,
  name: "IVAN",
  isActive: true,
  isBlocked: false,
};
export const testVipUser: IUser = {
  userId: 3,
  vip: true,
  name: "OLEG",
  isActive: true,
  isBlocked: false,
};

export const testRoomUser: RoomPlayer = {
  playerRoomStatus: PlayerRoomStatus.ACITVE,
  userId: 4,
  vip: true,
  name: "MISHA",
  isActive: true,
  isBlocked: false,
};
