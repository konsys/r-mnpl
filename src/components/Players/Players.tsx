import React from "react";
import { Avatar } from "../Avatar/Avatar";
import nanoid from "nanoid";

export interface IUser {
  userId: number;
  vip: boolean;
  registrationType?: string;
  name: string;
  avatar?: string;
  createdAt?: Date;
  updatedAt?: Date;
  isActive: boolean;
  isBlocked: boolean;
}
interface Prop {
  users: IUser[];
}

export const Players = (prop: Prop) => (
  <>
    <div key={nanoid(4)} className="table-body-players">
      {prop.users.map((v: IUser, k: number) => {
        const actionPlayer = k === 1 ? "1" : "0";
        return (
          <div
            key={nanoid(4)}
            className="table-body-players-card"
            id={"player_card_" + v.userId}
            mnpl-order="0"
            mnpl-team="undefined"
            mnpl-action_player={actionPlayer}
          >
            <Avatar
              key={nanoid(4)}
              name={v.name}
              money={15000}
              remainTime={53}
              img={v.avatar ? v.avatar : ""}
              isVip={v.vip}
            />

            <div className="table-body-players-card-menu">
              <div className="_profile" />
              <div className="_ignore" />
              <div className="_report" />
            </div>
          </div>
        );
      })}
    </div>
  </>
);
