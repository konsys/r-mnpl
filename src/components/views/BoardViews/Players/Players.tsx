import { IPlayer, IUser } from "../../../../types/types";
import React, { useEffect } from "react";
import {
  closePlayerActionEvent,
  openPlayerActionEvent,
  playerActionStore,
} from "../../../../stores/Board/PlayersStore";

import { Avatar } from "../Avatar/Avatar";
import { ICurrentAction } from "../../../../stores/Board/ActionStore";
import { PlayerActions } from "./PlayerActions/PlayerActions";
import { closeFieldActionEvent } from "../../../../stores/Board/FieldsStore";
import { useStore } from "effector-react";

interface Prop {
  players: IPlayer[];
  user: IUser;
  action: ICurrentAction;
}

export const Players = ({ players, user, action }: Prop) => {
  const actionStore = useStore(playerActionStore);

  const closePlayerAction = (event: any) => {
    (!event.target && !event.target.id && closePlayerActionEvent()) ||
      (event.target.id &&
        !(event.target.id.indexOf("player_card") > -1) &&
        closePlayerActionEvent());
  };

  useEffect(() => {
    document.addEventListener("click", closePlayerAction, false);
    return () => {
      document.removeEventListener("click", closePlayerAction, false);
    };
  }, []);

  return (
    <>
      <div className="table-body-players">
        {players &&
          players.map((player: IPlayer, index) => {
            return (
              <div
                key={index}
                className="table-body-players-card"
                id={"player_card_" + player.userId}
                mnpl-order={player.moveOrder}
                mnpl-team={player.team}
                mnpl-action_player={
                  player &&
                  action.event.action &&
                  action.event.action.userId === player.userId
                    ? 1
                    : 0
                }
                mnpl-opened={
                  actionStore.isVisible &&
                  actionStore.toUserId === player.userId
                    ? actionStore.position * 1
                    : 0
                }
                onClick={() => {
                  closeFieldActionEvent();
                  const user1 = (user && user.userId) || player.userId;
                  const user2 = player.userId;

                  return openPlayerActionEvent({
                    fromUserId: user1,
                    toUserId: user2,
                    position: 1,
                    isVisible: !actionStore.isVisible,
                    ignore: user1 !== user2,
                    ignoreOff: user1 !== user2,
                    profile: user1 !== user2,
                    contract: user1 !== user2,
                    creditTake: user1 === user2,
                    creditPay: false,
                    kick: user1 !== user2,
                    leave: user1 === user2,
                    report: user1 !== user2,
                    restart: user1 === user2,
                  });
                }}
              >
                <Avatar
                  key={index}
                  name={player.name}
                  money={player.money}
                  remainTime={53}
                  avatar={player.avatar ? player.avatar : ""}
                  isVip={player.vip}
                />
                <PlayerActions {...actionStore} />
              </div>
            );
          })}
      </div>
    </>
  );
};
