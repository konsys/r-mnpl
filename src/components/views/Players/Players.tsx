import React, { useEffect } from "react";
import {
  closePlayerActionEvent,
  openPlayerActionEvent,
  playerActionStore,
} from "../../../stores/PlayersStore";

import { Avatar } from "../Avatar/Avatar";
import { IPlayer } from "../../../types/types";
import { PlayerActions } from "./PlayerActions/PlayerActions";
import { actionsStore } from "../../../stores/ActionStore";
import { closeFieldActionEvent } from "../../../stores/FieldsStore";
import { useStore } from "effector-react";
import { userStore } from "../../../stores/UserStore";

interface Prop {
  players: IPlayer[];
}

export const Players = (prop: Prop) => {
  const action = useStore(actionsStore);
  const user = useStore(userStore);
  const actionStore = useStore(playerActionStore);

  const closePlayerAction = (event: any) => {
    (!event.target && !event.target.id && closePlayerActionEvent()) ||
      (event.target.id &&
        !(event.target.id.indexOf("player_card") > -1) &&
        closePlayerActionEvent());
  };

  useEffect(() => {
    document.addEventListener("click", closePlayerAction, false);
    return () => {
      document.removeEventListener("click", closePlayerAction, false);
    };
  }, []);

  return (
    <>
      <div className="table-body-players">
        {prop.players &&
          prop.players.map((player: IPlayer, index) => {
            return (
              <div
                key={index}
                className="table-body-players-card"
                id={"player_card_" + player.userId}
                mnpl-order={player.moveOrder}
                mnpl-team={player.team}
                mnpl-action_player={
                  player &&
                  action.event.action &&
                  action.event.action.userId === player.userId
                    ? 1
                    : 0
                }
                mnpl-opened={
                  actionStore.isVisible &&
                  actionStore.toUserId === player.userId
                    ? actionStore.position * 1
                    : 0
                }
                onClick={() => {
                  closeFieldActionEvent();
                  const user1 = (user && user.userId) || player.userId;
                  const user2 = player.userId;

                  return openPlayerActionEvent({
                    fromUserId: user1,
                    toUserId: user2,
                    position: 1,
                    isVisible: !actionStore.isVisible,
                    ignore: user1 !== user2,
                    ignoreOff: user1 !== user2,
                    profile: user1 !== user2,
                    contract: user1 !== user2,
                    creditTake: user1 === user2,
                    creditPay: false,
                    kick: user1 !== user2,
                    leave: user1 === user2,
                    report: user1 !== user2,
                    restart: user1 === user2,
                  });
                }}
              >
                <Avatar
                  key={index}
                  name={player.name}
                  money={player.money}
                  remainTime={53}
                  avatar={player.avatar ? player.avatar : ""}
                  isVip={player.vip}
                />
                <PlayerActions {...actionStore} />
              </div>
            );
          })}
      </div>
    </>
  );
};
