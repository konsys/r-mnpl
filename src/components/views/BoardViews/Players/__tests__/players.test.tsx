import { testPlayer1, testPlayer2, testUser } from "testMocks/user";

import { Avatar } from "../../Avatar/Avatar";
import { Players } from "../Players";
import React from "react";
import { shallow } from "enzyme";
import { testDoNothingAction } from "testMocks/action";

describe("Room top five test", () => {
  it("should render", () => {
    expect(
      shallow(
        <Players
          players={[testPlayer1, testPlayer2]}
          user={testUser}
          action={testDoNothingAction}
        />
      )
    ).toMatchSnapshot();
  });

  it("should render all players", () => {
    expect(
      shallow(
        <Players
          players={[testPlayer1, testPlayer2]}
          user={testUser}
          action={testDoNothingAction}
        />
      ).find(Avatar)
    ).toHaveLength(2);
  });
});
