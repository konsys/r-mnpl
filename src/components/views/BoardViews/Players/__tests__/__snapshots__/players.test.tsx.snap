// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Room top five test should render 1`] = `ShallowWrapper {}`;

exports[`Room top five test should render all players 1`] = `ShallowWrapper {}`;
